README.md

```diff
@@ -208,6 +213,183 @@ try {
+	receivingMessageTracer.start();
+	try {
+		// blocking call - until message is being available:
+		Message queryMessage = theQueue.receive("client queries");
+		ProcessingMessageTracer processingMessageTracer = oneAgentSDK
+			.traceProcessingMessage(messagingSystemInfo);
+		processingMessageTracer.setDynatraceStringTag(
+			queryMessage.getHeaderField(OneAgentSDK.DYNATRACE_MESSAGING_HEADERNAME));
+		processingMessageTracer.setVendorMessageId(queryMessage.msgId);
+		processingMessageTracer.setCorrelationId(queryMessage.correlationId);
+		processingMessageTracer.start();
+		try {
+			// do the work ... 
+		} catch (Exception e) {
+			processingMessageTracer.error(e.getMessage());
+			Logger.logError(e);
+		} finally {
+			processingMessageTracer.end();
+		}
+	} catch (Exception e) {
+		receivingMessageTracer.error(e.getMessage());
+		Logger.logError(e);
+	} finally {
+		receivingMessageTracer.end();
+	}
+}
+```
+
+In case of no blocking call happens (e. g. message processing via eventhandler), there is no need to use ReceivingMessageTracer - just trace processing of the message by using the ProcessingMessageTracer:
+
+```Java
+public void onMessage(Message message) {
+	MessagingSystemInfo messagingSystemInfo = oneAgentSDK.createMessagingSystemInfo("myMessagingSystem",
+		"requestQueue", MessageDestinationType.QUEUE, ChannelType.TCP_IP, "localhost:4711");
+
+	ProcessingMessageTracer processingMessageTracer = oneAgentSDK
+		.traceProcessingMessage(messagingSystemInfo);
+	processingMessageTracer.setDynatraceStringTag((String)
+		message.getObjectProperty(OneAgentSDK.DYNATRACE_MESSAGING_HEADERNAME));
+	processingMessageTracer.setVendorMessageId(queryMessage.msgId);
+	processingMessageTracer.setCorrelationId(queryMessage.correlationId);
+	processingMessageTracer.start();
+	try {
+		// do the work ... 
+	} catch (Exception e) {
+		processingMessageTracer.error(e.getMessage());
+		Logger.logError(e);
+	} finally {
+		processingMessageTracer.end();
+	}
+}
+```
+
+<a name="scav"/>
+
+## Add custom request attributes
+
+You can use the SDK to add custom request attributes to current traced service. Custom request attributes allows you to do easier/better filtering of your requests in dynatrace.
+
+Adding custom request attributes to the current traced service call is pretty simple. Just call one of the addCustomRequestAttribute methods with your key and value:
+
+```Java
+oneAgentSDK.addCustomRequestAttribute("region", "EMEA");
+oneAgentSDK.addCustomRequestAttribute("salesAmount", 2500);
+```
+
+When no service call is being traced, the custom request attributes are dropped. 
+
+<a name="limits"/>
+
+# Limits
+
+There are common limits, which will we applied when no other limit is explicitly mentioned.
+
+## String length
+
+There are two length limits for string parameters. Longer strings will be silently truncated:
+
+* SQL statements: default 4096 characters 
+* All other: 250 characters
```

Comment:
We have also one for WebRequest related strings. In code default is 250 but not sure if cluster sends on default something else.